src/transaction/build.rs

```diff
@@ -409,6 +469,82 @@ mod tests {
         assert_eq!(tx.gas_price, None);
     }
 
+    #[test]
+    fn tx_build_local_default_account_with_extra_gas_price() {
+        let mut transport = TestTransport::new();
+        let web3 = Web3::new(transport.clone());
+
+        let accounts = [
+            addr!("0x9876543210987654321098765432109876543210"),
+            addr!("0x1111111111111111111111111111111111111111"),
+            addr!("0x2222222222222222222222222222222222222222"),
+        ];
+
+        transport.add_response(json!(accounts)); // get accounts
+        transport.add_response(json!("0x42")); // gas price
+        let tx = BuildTransactionRequestForLocalSigningFuture::new(
+            &web3,
+            None,
+            GasPrice::Scaled(2.0),
+            TransactionRequestOptions::default(),
+        )
+        .immediate()
+        .expect("get accounts success");
+
+        transport.assert_request("eth_accounts", &[]);
+        transport.assert_request("eth_gasPrice", &[]);
+        transport.assert_no_more_requests();
+
+        assert_eq!(tx.from, accounts[0]);
+        assert_eq!(tx.gas_price, Some(U256::from(0x42 * 2)));
+    }
+
+    #[test]
+    fn tx_build_local_with_extra_gas_price() {
+        let mut transport = TestTransport::new();
+        let web3 = Web3::new(transport.clone());
+
+        let from = addr!("0xffffffffffffffffffffffffffffffffffffffff");
```

Comment:
Did you create an address macro? Or was this already available?

Reply:
I created it for unit tests. It’s in the `test_macros` module.